rfcs/055-prismic-api/README.md

```diff
@@ -0,0 +1,249 @@
+
+## Requirements, considerations, constraints
+
+- We'll be considering articles, exhibitions, and events in this first pass. Books won't be included for now, but could be added at a later date.
+- The API should only return enough information for users to figure out whether a result is relevant to them, and provide a link to the relevant page on wellcomecollection.org. The content of the pages themselves should still be fetched from Prismic directly.
+- Though we're describing them as a separate project, the new endpoints should seamlessly fit into the rest of the wellcomecollection.org API suite from a user POV. All of the existing conventions should be followed.
+- Despite prioritising search and defaulting to the Prismic API for display, We'll still include a way to fetch a single document by ID, eg `/articles/{id}`.
+- The API's URL structure should also be consistent with what appears on the front end of the site. For example, if article on the site appears at `/articles/{id}`, the API equivalent should be at `/articles/{id}`. The same should be true for search results.
+- The new API service should be written in typescript, following patterns set by the [concepts API](../050-concepts-api/README.md) for filtering, pagination, error handling, etc.
+- We shouldn't create new IDs for exhibitions, events, stories, etc. There aren't any situations where we should need to merge content from multiple sources, so we should use the document IDs directly from Prismic.
+- The elasticsearch index mapping should represent the contract between the pipeline and the API. The API shouldn't need to know anything about the structure of the data in Prismic.
+- We won't include an endpoint for `/series`, nor will we include series nodes in the `/articles` endpoint. If an article is part of a series, we'll denormalise information from its parent onto the article itself to use for relevance and visual signposting in results.
+
+## Proposed endpoints
+
+### `/articles`
+
+#### Example response
+
+```http
+GET /articles/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "body": "", // not exposed by default, should require an ?includes=body
+    "standfirst": "",
+    "published": "",
+    "contributors": [
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "partOf": [
+        {
+            ... // reuse the partOf model for series
+        }
+    ]
+}
+```
+
+```http
+GET /articles?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/articles?query=foo&filters=bar&sort=baz&page=2",
+}
+```
+
+#### Filters
+
+- date
+- contributor
+- series
+- contentType
```

Comment:
What is this?

Reply:
I think it's meant to be changed to `format` (webcomic, for example) after a chat we had yesterday, @harrisonpim right?

Reply:
```suggestion
- format
```